lib/ts/recipe/totp/types.ts

```diff
@@ -0,0 +1,197 @@
+    | {
+          type: "load";
+          showBackButton: boolean;
+          deviceInfo: TOTPDeviceInfo | undefined;
+          showAccessDenied: boolean;
+          error: string | undefined;
+          callingCreateDevice: boolean;
+      }
+    | {
+          type: "createDevice";
+          deviceInfo: TOTPDeviceInfo;
+      }
+    | {
+          type: "setBlocked";
+          error: string | undefined;
+          nextRetryAt: number;
+      }
+    | {
+          type: "setError";
+          showAccessDenied: boolean;
+          error: string | undefined;
+          maxAttemptCount?: number;
+          currAttemptCount?: number;
+      }
+    | {
+          type: "restartFlow";
+          error: string | undefined;
+      }
+    | {
+          type: "success";
+      }
+    | {
+          type: "showSecret";
+      };
+
+export type TOTPMFAState = {
+    deviceInfo?: TOTPDeviceInfo;
+    showSecret: boolean;
+    nextRetryAt?: number;
+    isBlocked: boolean;
+    showBackButton: boolean;
+    loaded: boolean;
+    error: string | undefined;
+    maxAttemptCount?: number;
+    currAttemptCount?: number;
+    showAccessDenied: boolean;
+};
+
+export type TOTPMFACommonProps = {
+    featureState: TOTPMFAState;
+    recipeImplementation: RecipeInterface;
+    config: NormalisedConfig;
+    userContext?: UserContext;
+};
+
+export type TOTPMFAProps = {
+    recipeImplementation: RecipeInterface;
+    config: NormalisedConfig;
+    onSuccess: () => void;
+    onShowSecretClicked: () => void;
+    onBackButtonClicked: () => void;
+    onRetryClicked: () => void;
+    onSignOutClicked: () => void;
+    dispatch: Dispatch<TOTPMFAAction>;
+    featureState: TOTPMFAState;
+    userContext?: UserContext;
+};
+export type TOTPMFAChildProps = Omit<TOTPMFAProps, "featureState" | "dispatch">;
+
+export type TOTPMFAScreenConfig = {
+    disableDefaultUI: boolean;
+    setupScreenStyle: string;
+    verificationScreenStyle: string;
+    blockedScreenStyle: string;
+    loadingScreenStyle: string;
+};
+
+export type UserInput = {
+    totpMFAScreen?: Partial<TOTPMFAScreenConfig>;
+
+    override?: {
+        functions?: (
+            originalImplementation: RecipeInterface,
+            builder?: OverrideableBuilder<RecipeInterface>
+        ) => RecipeInterface;
+    };
+} & RecipeModuleUserInput<GetRedirectionURLContext, PreAndPostAPIHookAction, OnHandleEventContext>;
+
+export type Config = UserInput &
+    RecipeModuleConfig<GetRedirectionURLContext, PreAndPostAPIHookAction, OnHandleEventContext>;
+
+export type NormalisedConfig = {
+    totpMFAScreen: TOTPMFAScreenConfig;
+
+    override: {
+        functions: (
+            originalImplementation: RecipeInterface,
+            builder?: OverrideableBuilder<RecipeInterface>
+        ) => RecipeInterface;
+    };
+} & NormalisedRecipeModuleConfig<GetRedirectionURLContext, PreAndPostAPIHookAction, OnHandleEventContext>;
+
+export type GetRedirectionURLContext =
+    | {
+          action: "MFA_TOTP";
```

Comment:
do we need this? Where is this being used? Cause the MFA recipe also has TO_FACTOR context type

Reply:
nope, removed.